spv/backend.go

```diff
@@ -56,17 +56,21 @@ type filterProof = struct {
 }
 
 // CFiltersV2 implements the CFiltersV2 method of the wallet.Peer interface.
+// This function blocks until a valid peer in the syncer returns the
+// appropriate CFilters.
 func (s *Syncer) CFiltersV2(ctx context.Context, blockHashes []*chainhash.Hash) ([]filterProof, error) {
 	for {
 		if err := ctx.Err(); err != nil {
 			return nil, err
 		}
-		rp, err := s.pickRemote(pickAny)
+		rp, err := s.waitForRemote(ctx, pickAny)
 		if err != nil {
 			return nil, err
 		}
 		fs, err := rp.CFiltersV2(ctx, blockHashes)
```

Comment:
We may want to leave another comment around here that if pickAny is changed to a different predicate, we would also need to check for `rp == nil`.

Reply:
I've changed the signature for `waitForRemote()` to pass an explicit flag to control whether to wait or not for at least one eligible peer. This should make it safer for callers.